### PR TITLE
Fixing Labels and weapon Initialisation

### PR DESCRIPTION
**Pull request by rohankumar17**
# Description

This update addresses an issue where the initial weapon counter incorrectly displayed zero, despite the player already having a weapon equipped. The weapon counter now properly reflects the initial weapon status. Additionally, the initial ammo count has been customized to vary according to different player types, ensuring correct ammo initialization. An issue with ammo shells appearing incorrectly on the map has also been resolved.

Fixes / Closes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

I extensively tested these changes by playing the game for several hours, identifying the issues, and ensuring the fixes worked correctly. After making the changes, I played the game again for an extended period to verify that the weapon counter and ammo behavior functioned as expected, and that the ammo shells no longer appeared incorrectly on the map.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by IanP96** [comment]
there are merge conflicts

File: source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java [hunk 1]
```diff
@@ -58,12 +57,42 @@ public void create() {
...
+                playerNum == PlayerAnimationController.PlayerNum.PLAYER_2 ||
+                playerNum == PlayerAnimationController.PlayerNum.PLAYER_3 ||
+                playerNum == PlayerAnimationController.PlayerNum.BEAR) {
+            // Players 2, 3, 4, and Bear have 7 ammo with the gun
+            updateGunLabel(7, "Gun");
+            updateRangedWeaponUI(7, "Gun");
+        } else {
+            // For other players, set the gun to have 20 ammo
+            updateGunLabel(20, "Gun");
+            updateRangedWeaponUI(20, "Gun");
```
Comment (manya-k) [inline source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java:94]: the padding hasn't been configured properly. it is making the goblin statue render improperly:
image

**Comment by vsinh14** [comment]
This PR is breaking the UI of items 
image

**Review by manya-k — APPROVED** [review]
the game crashes if you go to the second room and killl all animals.
it crashes at spawn method

**Comment by rohankumar17** [comment]
Ok working on it might have been some issue during merging

**Comment by rohankumar17** [comment]
The game is not crashing because of my updates

**Pull request closed without merging**